src/web/mod.rs

```diff
@@ -110,9 +110,13 @@ impl Handler for CratesfyiHandler {
             })
             .or_else(|e| {
                 debug!("{}", e.description());
-                let err: error::Nope = *e.error
-                    .downcast::<error::Nope>()
-                    .expect("all cratesfyi errors should be of type Nope");
+                let err = if let Some(err) = e.error.downcast::<error::Nope>() {
+                    *err
+                } else if e.error.downcast::<NoRoute>().is_some() {
+                    error::Nope::ResourceNotFound
+                } else {
+                    panic!("all cratesfyi errors should be of type Nope");
```

Comment:
I've poked about a bit in Iron and, in addition to `NoRoute`, the router may also give back a [`TrailingSlash`](http://ironframework.io/doc/src/router/src/router.rs.html#212-225). This happens if you try to access `/releases/` instead of the valid `/releases` and will normally redirect you to the right path.

Doing something like this:

```rust
 let err = if let Some(err) = e.error.downcast::<error::Nope>() {
    *err
} else if e.error.downcast::<NoRoute>().is_some() {
    error::Nope::ResourceNotFound
} else if e.error.downcast::<TrailingSlash>().is_some() {
    // We want to redirect to the right path, so just return the response
    return Ok(e.response);
} else {
    unreachable!("All cratefyi errors should be of type Nope");
};
Self::chain(err).handle(req)
```

makes that work again but I don't know whether returning early is the right thing to do. Don't see why it'd be a problem but I don't know that much about Iron.